project_setup/branch-protection.md

```diff
@@ -0,0 +1,36 @@
+# Branch Protection #
+
+We require branch protection to be enabled on all cisagov repositories after
+creation and initial population.
+
+## Setting up a rule to protect the `develop` branch ##
+
+To enable branch protection, you must have permission to access the Settings
+menu.
+
+In `Settings`, go to the `Branches` entry and create a rule with the following:
+
+- Branch name pattern: `develop`
+  - Note: Our primary branches are named `develop`, as a rule.
+  - Verify that adding this name pattern then shows that it
+`Applies to 1 branch: develop`.
+
+- Protect matching branches
+  - [X] Require pull request reviews before merging
+    - Required approving reviews: `2`
+    - [ ] Dismiss stale pull request approvals when new commits are pushed
+    - [X] Require review from Code Owners
+    - [ ] Restrict who can dismiss pull request reviews
```

Comment:
Historically we have always checked this box, I believe.

Reply:
This was based on an existing repo - while I haven't done an exhaustive survey, both `cool-system` and `development-guide` don't have it checked, while `ansible-role-cobalt-strike` and `teamserver-packer` do have it checked.

I'm in favor of standardizing on whatever the team prefers.

Reply:
I prefer that we do restrict who can dismiss PR reviews.

Reply:
I agree with @dav3r that we should restrict who can dismiss PR reviews.  Checking this box gets rid of the possibility that someone simply dismisses a request for changes that he or she disagrees with instead of hashing out the issue with the reviewer in the PR.

Reply:
Resolved in https://github.com/cisagov/development-guide/pull/44/commits/18a952c15253fae65b5c657a3bea687c56f4fdcb